Tests/BasicsTests/HTTPClientTests.swift

```diff
@@ -219,13 +231,14 @@ final class HTTPClientTests: XCTestCase {
 
         do {
```

Comment:
Consider replacing this `do`/`catch` with `#expect(throws: HTTPClientError.badResponseStatusCode(statusCode)) { ... }`